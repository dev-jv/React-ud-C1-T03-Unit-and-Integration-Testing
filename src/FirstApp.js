import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ col, esp }) => {
        return (
            <>
                <h1>{ col }</h1>
                <p>{ esp }</p>
            </>
        );
};

FirstApp.propTypes = {
    col : PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
    esp: 'Kaede',
};

export default FirstApp;
