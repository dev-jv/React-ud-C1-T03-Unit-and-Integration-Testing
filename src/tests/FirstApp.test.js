import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import FirstApp from "../FirstApp";
import { shallow } from 'enzyme';
import "../setupTests"
import colors from 'colors';

describe('/> Testing in <FirstApp />'.bgBlue.white, () => {

    // ------------------------------------- <> with render...
    // test('Test 1: Should define the red color', () => {
    //     const col = 'Red';
    //     const { getByText } = render(<FirstApp col={col}/>);
    //     expect( getByText(col + '') ).toBeInTheDocument();
    //     // const wrapper = render(<FirstApp col={col}/>);
    //     // expect( wrapper.getByText(col) ).toBeInTheDocument();
    // })

    // ------------------------------------- <> with Enzyme
    test('Test 1: Should show <FirstApp/> correctly', () => {
        const col = 'Red';
        const wrapper = shallow( <FirstApp col={ col }/> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('Test 2: Should show the species sent by props', () => {
        const col = 'Red';
        const esp = 'Miyasama Kaede';
        const wrapper = shallow(
            <FirstApp
                col={ col }
                esp={ esp }
            />
        );
        const txt = wrapper.find('p').text();
        expect(txt).toBe(esp)
    })

});
