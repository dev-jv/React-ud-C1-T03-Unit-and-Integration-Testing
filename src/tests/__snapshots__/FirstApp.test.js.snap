// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[37m[44m/> Testing in <FirstApp />[49m[39m Test 1: Should show <FirstApp/> correctly 1`] = `
<Fragment>
  <h1>
    Red
  </h1>
  <p>
    Kaede
  </p>
</Fragment>
`;
